src/clients/UBAClient/UBAClientUtilities.ts

```diff
@@ -401,19 +401,21 @@ async function getFlows(
     return validWithinBounds || hasMatchingDeposit;
   });
 
-  const refundRequests: UbaFlow[] = spokePoolClient.getRefundRequests(fromBlock, toBlock).filter((refundRequest) => {
-    const result = refundRequestIsValid(chainId, chainIdIndices, spokePoolClients, hubPoolClient, refundRequest);
-    if (!result.valid && logger !== undefined) {
-      logger.info({
-        at: "UBAClient::getFlows",
-        message: `Excluding RefundRequest on chain ${chainId}`,
-        reason: result.reason,
-        refundRequest,
-      });
-    }
+  const refundRequests: UbaFlow[] = spokePoolClient
+    .getRefundRequests(fromBlock, toBlock)
+    .filter(async (refundRequest) => {
+      const result = await refundRequestIsValid(chainIdIndices, spokePoolClients, hubPoolClient, refundRequest);
+      if (!result.valid && logger !== undefined) {
+        logger.info({
+          at: "UBAClient::getFlows",
+          message: `Excluding RefundRequest on chain ${chainId}`,
+          reason: result.reason,
+          refundRequest,
+        });
+      }
 
-    return result.valid;
-  });
+      return result.valid;
+    });
```

Comment:
We're going to need to modify this to get the output the way we want.

```
  const refundRequests: UbaFlow[] = spokePoolClient
    .getRefundRequests(fromBlock, toBlock)
    .filter(async (refundRequest) => {
      const result = await refundRequestIsValid(chainIdIndices, spokePoolClients, hubPoolClient, refundRequest);
      if (!result.valid && logger !== undefined) {
        logger.info({
          at: "UBAClient::getFlows",
          message: `Excluding RefundRequest on chain ${chainId}`,
          reason: result.reason,
          refundRequest,
        });
      }

      return result.valid;
    });
```

should probably become

```
  const refundRequests: UbaFlow[] = await filterAsync(
    spokePoolClient.getRefundRequests(fromBlock, toBlock),
    async (refundRequest) => {
      const result = await refundRequestIsValid(chainIdIndices, spokePoolClients, hubPoolClient, refundRequest);
      if (!result.valid && logger !== undefined) {
        logger.info({
          at: "UBAClient::getFlows",
          message: `Excluding RefundRequest on chain ${chainId}`,
          reason: result.reason,
          refundRequest,
        });
      }
      return result.valid;
    });
```